Tests/TuistLoaderIntegrationTests/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilderIntegrationTests.swift

```diff
@@ -113,3 +158,85 @@ final class ProjectDescriptionHelpersBuilderIntegrationTests: TuistTestCase {
         XCTAssertTrue(exists)
     }
 }
+
+private class MockFileHandling: FileHandler {
+    var createFolderCount = 0
+    override public func createFolder(_: Path.AbsolutePath) throws {
+        createFolderCount += 1
+    }
+}
+
+private class MockFileSystem: FileSysteming {
```

Comment:
I would avoid creating a custom mock.

If you feel strongly about mocking the file system, it should be done with [Mockable](https://github.com/Kolos65/Mockable).

Reply:
Are we ok with incorporating the Mockable framework within the FileSystem SPM? I can create a PR supporting it, and then update this one to incorporate the change.
I do wonder on the choice to make the protocol conformance a struct? Should we make it a class that can be overriden?

Reply:
We can put it in the `FileSystem`, but I do wonder if we need to have the extra verification here – checking whether `createFolder` is called should be an implementation detail.

Reply:
yep, you are right, see my other comment regarding the integration tests